usr.bin/mdo/mdo.c

```diff
@@ -22,10 +23,29 @@ usage(void)
 	exit(EXIT_FAILURE);
 }
 
+static char*
+join_strings(char **arr, int size, const char *delimiter)
+{
+	int total_length = 0, delimiter_length = strlen(delimiter);
```

Comment:
There are several branches in these loops, also I think it's a good idea to print the function name where it fails.
As suggested by [delphij](https://github.com/freebsd/freebsd-src/pull/1512/files#r1845332811), you can also use sbuf(9) so that you can probably eliminate this wrapper function.

Slightly changed
```c
static char *join_strings2(char **arr, int nlen, const char *delim)
{
	char *buf;
	size_t dlen, tlen, i;

	tlen = 0;
	dlen = strlen(delim);
	for (i = 0; i < nlen; i++)
		tlen += strlen(arr[i]) + dlen;

	if ((buf = malloc(tlen + 1)) == NULL)
		err(1, "malloc()");

	*buf = '\0';
        strcat(buf, arr[0]);
	for (i = 1; i < nlen; i++) {
		strcpy(buf + strlen(buf), delim);
		strcpy(buf + strlen(buf), arr[i]);
	}

	return (buf);
}
```